CMakeLists.txt

```diff
@@ -74,12 +74,15 @@ if ("${CMAKE_CXX_COMPILER_ID}" MATCHES "^(AppleClang|Clang|GNU)$")
   add_flag(-Wno-format-nonliteral)   # prints way too many warnings from spdlog
   add_flag(-Wno-gnu-zero-variadic-macro-arguments)   # https://stackoverflow.com/questions/21266380/is-the-gnu-zero-variadic-macro-arguments-safe-to-ignore
 
-  # promote to errors
-  add_flag(-Werror-unused-lambda-capture)  # error if lambda capture is unused
-  add_flag(-Werror-return-type)      # warning: control reaches end of non-void function [-Wreturn-type]
-  add_flag(-Werror-non-virtual-dtor) # warn the user if a class with virtual functions has a non-virtual destructor. This helps catch hard to track down memory errors
-  add_flag(-Werror-sign-compare)     # warn the user if they compare a signed and unsigned numbers
-  add_flag(-Werror-reorder)          # field '$1' will be initialized after field '$2'
+  # clang compiler starting from version 12 wont compile when these unknown for it flags are specified
+  if (NOT ("${CMAKE_CXX_COMPILER_ID}" MATCHES "^(AppleClang|Clang)$") AND (${CMAKE_CXX_COMPILER_VERSION} VERSION_GREATER_EQUAL "12.0"))
```

Comment:
check again, I think negation is applied only to the first term

Reply:
I think it is better not  to ignore, but set proper analogous warnings for clang-12

Reply:
Added required parenthesis.

Reply:
Will recheck analogues for clang 12